tools/network.py

```diff
@@ -83,7 +154,39 @@ def fetch_raw(
     request_blocking = QgsBlockingNetworkRequest()
     if authcfg_id:
         request_blocking.setAuthCfg(authcfg_id)
-    _ = request_blocking.get(req)
+    # QNetworkRequest *only* supports get and post. No idea why.
+    if method == "get":
+        _ = request_blocking.get(req)
+    elif method == "post":
+        if data:
+            # Support JSON
+            byte_data = bytes(json.dumps(data), encoding)
+            req.setRawHeader(b"Content-Type", bytes("application/json", encoding))
```

Comment:
The used encoding should be added to the header like `application/json; charset=utf-8`